tests/gui/components/update-notifier.spec.js

```diff
@@ -1,148 +1,203 @@
+            lastSleptUpdateNotifier: undefined,
+            releaseType: release.RELEASE_TYPE.UNKNOWN
+          });
 
-            beforeEach(function() {
-              settings.set('lastUpdateNotify', undefined);
-            });
+          m.chai.expect(result).to.be.true;
+        });
 
-            it('should return true', function() {
-              const result = UpdateNotifierService.shouldCheckForUpdates({
-                ignoreSleepUpdateCheck: this.ignoreSleepUpdateCheck
-              });
+      });
 
-              m.chai.expect(result).to.be.true;
-            });
+      describe('given the `lastSleptUpdateNotifier` was very recently updated', function() {
 
+        it('should return false if releaseType is production', function() {
+          const result = UpdateNotifierService.shouldCheckForUpdates({
+            lastSleptUpdateNotifier: Date.now() - 1000,
+            releaseType: release.RELEASE_TYPE.PRODUCTION
           });
 
-          describe('given the `lastUpdateNotify` was very recently updated', function() {
-
-            beforeEach(function() {
-              settings.set('lastUpdateNotify', Date.now() + 1000);
-            });
+          m.chai.expect(result).to.be.false;
+        });
 
-            it('should return false', function() {
-              const result = UpdateNotifierService.shouldCheckForUpdates({
-                ignoreSleepUpdateCheck: this.ignoreSleepUpdateCheck
-              });
+        it('should return true if releaseType is snapshot', function() {
+          const result = UpdateNotifierService.shouldCheckForUpdates({
+            lastSleptUpdateNotifier: Date.now() - 1000,
+            releaseType: release.RELEASE_TYPE.SNAPSHOT
+          });
 
-              m.chai.expect(result).to.be.false;
-            });
+          m.chai.expect(result).to.be.true;
+        });
 
+        it('should return true if releaseType is unknown', function() {
+          const result = UpdateNotifierService.shouldCheckForUpdates({
+            lastSleptUpdateNotifier: Date.now() - 1000,
+            releaseType: release.RELEASE_TYPE.UNKNOWN
           });
 
-          describe('given the `lastUpdateNotify` was updated long ago', function() {
+          m.chai.expect(result).to.be.true;
+        });
 
-            beforeEach(function() {
-              const SLEEP_MS = units.daysToMilliseconds(UPDATE_NOTIFIER_SLEEP_DAYS);
-              settings.set('lastUpdateNotify', Date.now() + SLEEP_MS + 1000);
-            });
+      });
 
-            it('should return true', function() {
-              const result = UpdateNotifierService.shouldCheckForUpdates({
-                ignoreSleepUpdateCheck: this.ignoreSleepUpdateCheck
-              });
+      describe('given the `lastSleptUpdateNotifier` was updated in the future', function() {
 
-              m.chai.expect(result).to.be.true;
-            });
+        it('should return false if releaseType is production', function() {
```

Comment:
Hmmm, thinking about it, does it make sense for `shouldCheckForUpdates` to return `false` if `lastSleptUpdateNotifier` was updated in the future? Perhaps we should treat a future `lastSleptUpdateNotifier` as an 'error condition' and always return `true`?